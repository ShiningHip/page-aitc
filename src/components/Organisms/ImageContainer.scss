.container {
  width: 100%;
  height: 100%;

  &::before {
    content: '';
    display: block;
    padding-top: 120%;
  }
}

.info_list > *,
.track_list > * {
  -webkit-transition: 0.3s;
  transition: 0.3s;
  transition-delay: 0.2s;
  transition-timing-function: ease-in-out;
}

.info_eyecatch {
  left: 0;
  right: 0;
  background: url('/static/AITC-0001/logo.png') no-repeat;
  background-size: contain;
  background-position: center;
  margin: 0 auto;
  height: 200px;
  position: relative;
  display: block;
}

.info {
  height: 100%;
  padding-top: 40px;
  z-index: 20;
  overflow: hidden;
  position: relative;
  display: flex;
  background: rgba(0, 0, 0, 0.7);
}

.track {
  @extend .info;
}

.credit {
  @extend .info;
}

.info_container {
  width: 100%;
  height: 100%;
  position: absolute;
  text-align: center;
}

.track_container {
  @extend .info_container;
}

.credit_container {
  @extend .info_container;
}

.subtitle {
  width: auto;
  height: 40px;
  font-size: 22px;
  font-weight: 400;
  line-height: 40px;
  right: 0;
  left: 0;
  text-align: center;
  position: absolute;
  color: #fdfdfd;

  &::after {
    content: '';
    margin-top: -5px;
    width: 100vw;
    height: 3px;
    display: block;
    background: #f0f0f0;
  }
}

ul.info_list,
ul.track_list,
ul.credit_list {
  top: 70px;
  list-style: none;
  overflow: hidden;
  width: 90%;
  margin: 0 auto;
  position: relative;
  display: block;
}

@media screen and (min-width: 1280px) {
  .subtitle {
    color: #fdfdfd;
  }

  .info,
  .track,
  .credit {
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
  }

  .credit_container,
  .info_container,
  .track_container {
    right: 0;
    left: 0;
    margin: auto;
  }
}

@media screen and (max-width: 1280px) {
  .subtitle {
    color: #fdfdfd;

    &::after {
      background: #f0f0f0;
    }
  }

  .credit_container,
  .info_container,
  .track_container {
    width: 100%;
  }

  .info_eyecatch {
    width: 100%;
    background-position: center;
  }
}
