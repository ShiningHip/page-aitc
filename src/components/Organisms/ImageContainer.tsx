import React from 'react'
import style from './ImageContainer.scss'

interface Props {
  containerType: string
  children: React.ReactNode
}

const ImageContainer: React.FC<Props> = (props: Props) => (
  <>
    <style jsx>{style}</style>
    <div className={`${props.containerType} container`}>
      <div className={`${props.containerType}_container`}>
        <ul className={`${props.containerType}_list`}>{props.children}</ul>
      </div>
    </div>
  </>
)

export default ImageContainer
