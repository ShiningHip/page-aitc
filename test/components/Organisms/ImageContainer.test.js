import ImageContainer from 'components/Organisms/ImageContainer'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('ImageContainer test', () => {
  const props = {
    containerType: 'test'
  }

  it('表示が正しいこと', () => {
    const tree = shallow(<ImageContainer {...props} />).toJSON
    expect(tree).toMatchSnapshot()
  })
})
