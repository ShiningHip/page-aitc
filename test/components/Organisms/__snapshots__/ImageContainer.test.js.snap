// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer test 表示が正しいこと 1`] = `undefined`;
